### PR TITLE
feat: support defaultInternalBehavior in preview and playground plugin

### PR DESCRIPTION
**Pull request by kainstar**
## Summary

Support `defaultInternalBehavior` in preview and playground plugin.

## Related Issue

https://github.com/web-infra-dev/rspress/issues/275

## By the way

With this PR user can use preview and playground plugin both for internal component. And for better DX, I suggest use `preview` and `playground` tag instead of current `code`, then user can manullay control how external component should be transformer, and `code` tag is changed to a simple way to render as normal code block.

If you agree with me and want more discussions, we can create an other new issue or discussion.

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [x] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by sanyuan0704** [comment]
Why the name is `defaultInternalBehavior` instead of `defaultBehavior`? I think the latter is easier.

File: packages/document/docs/en/plugin/official-plugins/playground.mdx [hunk 1]
```diff
@@ -82,12 +82,21 @@ export default App;
 ```
 ````
 
+If you have set `defaultInternalBehavior` to `'pure'`, Rspress will not render this component by default, but instead display it as a code block. If you want to render a code block as a component in this situation, you can do add the `preview` identifier to specify, the usage is as follows:
+
+````md
```
Comment (sanyuan0704) [inline packages/document/docs/en/plugin/official-plugins/playground.mdx:87]: The language should be `mdx`

**Comment by kainstar** [comment]
> Why the name is `defaultInternalBehavior` instead of `defaultBehavior`? I think the latter is easier.

Because this option only decide how internal components render, if use `defaultBehavior`, it may give others the impression that it affects both internal and external components. If you perfer a shorter name, perhaps `internalBehavior` is also an option?

File: packages/document/docs/en/plugin/official-plugins/playground.mdx (hunk 1 above)
Comment (kainstar) [reply packages/document/docs/en/plugin/official-plugins/playground.mdx]: fixed

**Comment by sanyuan0704** [comment]
On the one hand, the default behavior can be both applied in internal and external components, which is able to be implemented. For <code /> tag, we can parse the prop `renderMode`, which may be `pure` or `preview`.

On the other hand, the param name can be `defaultRenderMode`. i think this name is better.

**Comment by sanyuan0704** [comment]
cc @kainstar

**Comment by kainstar** [comment]
I have renamed `defaultInternalBehavior` to `defaultRenderMode`

**Review by sanyuan0704 — APPROVED** [review]

**Pull request merged**